dataplane/dataplane.cpp

```diff
@@ -1969,9 +2094,55 @@ eResult cDataPlane::checkConfig()
 		}
 	}
 
+	if (!checkControlPlaneWorkersConfig())
+	{
+		return eResult::invalidConfigurationFile;
+	}
+
 	return eResult::success;
 }
 
+bool cDataPlane::checkControlPlaneWorkersConfig()
+{
+	std::set<InterfaceName> assigned;
+	std::set<InterfaceName> to_assign = workerInterfacesToService();
+	bool result = true;
+	for (const auto& [core, worker_ports] : config.controlplane_workers)
+	{
+		for (const auto& p : worker_ports)
+		{
+
+			if (assigned.find(p) != assigned.end())
+			{
+				YADECAP_LOG_ERROR("Duplicate port in control plane worker config: core: %d, port: %s\n", core, p.c_str());
+				result = false;
+				continue;
+			}
+
+			if (to_assign.find(p) == to_assign.end())
+			{
+				YADECAP_LOG_ERROR("Control plane worker config contains port not assigned to fast worker: core: %d, port: %s\n", core, p.c_str());
+				result = false;
+				continue;
+			}
+
+			assigned.emplace(p);
+			to_assign.erase(p);
```

Comment:
Instead of checking with `find` and then doing `emplace` and `erase`, we can just use return value of those methods:
```cpp
            if (!assigned.insert(p).second)
            {
                YADECAP_LOG_ERROR("Duplicate port in control plane worker config: core: %d, port: %s\n", core, p.c_str());
                result = false;
                continue;
            }

            if (to_assign.erase(p) == 0)
            {
                YADECAP_LOG_ERROR("Control plane worker config contains port not assigned to fast worker: core: %d, port: %s\n", core, p.c_str());
                result = false;
                continue;
            }
```

`insert` returns a pair containing an iterator to the inserted element and a flag indicating whether the insertion took place.
`erase` returns the number of elements removed, which is 0 if the element was not found.